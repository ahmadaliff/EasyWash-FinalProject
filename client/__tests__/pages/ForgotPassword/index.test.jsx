import { render, fireEvent, act } from '@utils/testHelper';
import ForgotPassword from '@pages/ForgotPassword';

let wrapper;
const mockDispatch = jest.fn();
const mockNavigate = jest.fn();
const mockLogin = false;

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockDispatch,
}));
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
  Link: jest.fn().mockImplementation(({ to, children }) => <a href={to}>{children}</a>),
}));
jest.mock('@containers/Client/selectors', () => ({
  selectLogin: () => mockLogin,
}));

beforeEach(() => {
  jest.clearAllMocks();
  act(() => {
    wrapper = render(<ForgotPassword />);
  });
});
describe('ForgotPassword Page', () => {
  test('Renders correctly', () => {
    const { getByText, getByLabelText, getByTestId } = wrapper;
    expect(getByLabelText('Email')).toBeInTheDocument();
    expect(getByTestId('button-submit')).toBeInTheDocument();
    expect(getByTestId('navigate-logo')).toBeInTheDocument();
    expect(getByText('Punya akun?')).toBeInTheDocument();
    expect(getByText('Masuk')).toBeInTheDocument();
  });

  test('Displays error message for invalid email', async () => {
    const { getByLabelText, getByTestId } = wrapper;
    const emailInput = getByLabelText('Email');
    fireEvent.change(emailInput, { target: { value: 'invalid-email' } });
    await act(() => {
      fireEvent.click(getByTestId('button-submit'));
    });
  });

  test('Submits form successfully and redirects to login', async () => {
    const { getByLabelText, getByTestId } = wrapper;
    const emailInput = getByLabelText('Email');
    fireEvent.change(emailInput, { target: { value: 'valid-email@example.com' } });
    await act(() => fireEvent.click(getByTestId('button-submit')));
    expect(mockDispatch).toHaveBeenCalledTimes(1);
  });
  test('Submits form successfully and redirects to login', async () => {
    const { getByTestId } = wrapper;
    const logo = getByTestId('navigate-logo');
    act(() => {
      fireEvent.click(logo);
    });
    expect(mockNavigate).toHaveBeenCalledWith('/');
  });
  test('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
