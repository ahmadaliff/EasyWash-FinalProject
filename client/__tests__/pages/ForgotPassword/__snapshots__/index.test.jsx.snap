// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword Page Matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="mainWrap"
      >
        <div
          class="logoHeader"
          data-testid="navigate-logo"
        >
          <img
            alt="logo"
            class="logo"
            src="longLogo.svg"
          />
        </div>
        <div
          class="forgotPasswordCard"
        >
          <h3
            class="forgotPasswordHeader"
          >
            Lupa Password
          </h3>
          <form
            class="form"
          >
            <div
              class="MuiFormControl-root inputContainer css-1nrlq1o-MuiFormControl-root"
              data-testid="inputRHF"
            >
              <label
                class="inputLabel"
                for="email"
              >
                Email
                <input
                  class="input undefined"
                  id="email"
                  name="email"
                  type="text"
                  value=""
                />
              </label>
            </div>
            <button
              class="buttonSubmit"
              data-testid="button-submit"
              disabled=""
              type="submit"
            >
              Lupa Password
            </button>
            <div
              class="formNav"
            >
              <p
                class="nav"
              >
                Punya akun? 
                <a
                  href="/login"
                >
                  Masuk
                </a>
              </p>
            </div>
          </form>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="mainWrap"
    >
      <div
        class="logoHeader"
        data-testid="navigate-logo"
      >
        <img
          alt="logo"
          class="logo"
          src="longLogo.svg"
        />
      </div>
      <div
        class="forgotPasswordCard"
      >
        <h3
          class="forgotPasswordHeader"
        >
          Lupa Password
        </h3>
        <form
          class="form"
        >
          <div
            class="MuiFormControl-root inputContainer css-1nrlq1o-MuiFormControl-root"
            data-testid="inputRHF"
          >
            <label
              class="inputLabel"
              for="email"
            >
              Email
              <input
                class="input undefined"
                id="email"
                name="email"
                type="text"
                value=""
              />
            </label>
          </div>
          <button
            class="buttonSubmit"
            data-testid="button-submit"
            disabled=""
            type="submit"
          >
            Lupa Password
          </button>
          <div
            class="formNav"
          >
            <p
              class="nav"
            >
              Punya akun? 
              <a
                href="/login"
              >
                Masuk
              </a>
            </p>
          </div>
        </form>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
